core/sds-aspect-meta-model-java/src/main/java/io/openmanufacturing/sds/metamodel/loader/MetaModelBaseAttributes.java

```diff
@@ -219,6 +219,31 @@ private static String getSyntheticName( final Resource modelElement, final Model
       return parentModelElementName + modelElementTypeName;
    }
 
+   // We have to be careful when searching for the parent nodes with a regular name - the "listStatements" API returns the matching nodes
+   // in no particular order; with some very specific models this could lead to non-deterministic behavior.
+   // In the following very simplified example we are looking for ":NumberList" as the parent of "_:blankNode", but could get the anonymous node [] instead.
+   // [
+   //  aux:contains _:blankNode ;
+   // ] .
+   // :NumberList a bamm-c:List ;
+   //    bamm-c:elementCharacteristic _:blankNode .
+   // _:blankNode a bamm-c:Trait ;
+   private static Resource getNamedParent( final Resource modelElement, final Model model ) {
```

Comment:
Please add unit test(s) that verify the correct behaviour.

Reply:
Done